### ISSUE TITLE
SharedMappings ignores multiple tsconfig.json paths

### ISSUE THREAD

**Issue by glebmachine** [angular-architects/module-federation-plugin#44]
By design in tsconfig.json you able to set multilple paths like so:
```
"@wololo/auth": [
  "dist/@wololo/auth", 
  "libs/auth/src/index"
]
```

This declaration works by following idea: "use compiled library, if not - compile from source"

But, here as you can see you just pick first path:
https://github.com/angular-architects/module-federation-plugin/blob/c60bf4f0878ba6c9bf0a3b0ce0baf1acb93226e1/packages/mf/src/utils/shared-mappings.ts#L33

**Comment by manfredsteyer** [comment]
Yes, this is correct. Nx mono repo are the role model here. They just define one path.

**Comment by glebmachine** [comment]
This is bad(

In my project i'm compile monorepo libraries and use artefacts for faster build/serve in development mode
Multiple pass allow me to enumerate all paths available for my libraries (first - compiled path, secondary - source path as shown in case below)

**Comment by glebmachine** [comment]
"Nx mono repo are the role model here. They just define one path." i think thet define only one path, but doesn't restrict user to do that. So, seems fair to support this feature (because of tsconfig specification as well)

**Comment by oggythorpe1610** [comment]
Having been working on a project myself, I agree this is an issue. I had a definition in my tsconfig.json file for my shared library below two other **'dist/ ...'** entries. in the paths:{} section for the two libraries I had created. Everything broke saying it couldn't find the modules despite having added the path of **'projects/<my-lib-name>/src/public-api.ts'** to the end of the list. Only moving them to the very first entry fixed things and everything worked as it should. Either this needs to be documented to add your project library path reference as the first entry above any **'dist/<my-lib-name> ...'** or the code needs to account for multiple entries correctly.

Obviously <my-lib-name> is replaced by the library you have yourself here :)

**Comment by shankaryadavy9** [comment]
Hi @manfredsteyer
Is there any update on this..?.  Having support for multiple entries will definitely help everyone.
I even explained my scenario  on this PR(Let me know if there is another way to achieve it.)
https://github.com/angular-architects/module-federation-plugin/pull/52#issuecomment-869106938
